use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/domain/geo/DistanceCalculator.java

```diff
@@ -0,0 +1,48 @@
+package org.acme.vehiclerouting.domain.geo;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.acme.vehiclerouting.domain.Location;
+
+public interface DistanceCalculator {
+
+    /**
+     * Calculate the distance between {@code from} and {@code to} in meters or driving time seconds when using {@link DrivingTimeDecorator} class.
+     *
+     * @param from starting location
+     * @param to target location
+     * @return meters or driving time in seconds, depending on the implementation
+     */
+    long calculateDistance(Location from, Location to);
+
+    /**
+     * Bulk calculation of distance.
+     * Typically, much more scalable than {@link #calculateDistance(Location, Location)} iteratively.
+     *
+     * @param fromLocations never null
+     * @param toLocations never null
+     * @return never null
+     */
+    default Map<Location, Map<Location, Long>> calculateBulkDistance(
+            Collection<Location> fromLocations,
+            Collection<Location> toLocations) {
+        return fromLocations.stream().collect(Collectors.toMap(
+                Function.identity(),
+                from -> toLocations.stream().collect(Collectors.toMap(
+                        Function.identity(),
+                        to -> calculateDistance(from, to)))));
+    }
+
+    /**
+     * Calculate distance matrix for the given list of locations and assign distance maps accordingly.
+     *
+     * @param locations locations list
+     */
+    default void initDistanceMaps(Collection<Location> locations) {
+        Map<Location, Map<Location, Long>> distanceMatrix = calculateBulkDistance(locations, locations);
+        locations.forEach(location -> location.setDrivingTimeSeconds(distanceMatrix.get(location)));
```

Comment:
The `Location` uses seconds, so here, in this method, we already decide it's seconds and not meters.

What if we did this initialization in the `VehicleRoutePlan` and left the `DistanceCalculator` to only calculate distance?

Reply:
Good point! Another possibility which just occurred to me could be passing the setter method as a parameter, but that could be less readable.

Reply:
As agreed, I have simplified the quickstart to work with travel time only (not with distance).